docs/explanation/index.rst

```diff
@@ -9,4 +9,5 @@ This section covers conceptual questions about Testflinger.
     agents
     queues
     job-priority
+    restricted-queues
```

Comment:
This is referenced in the index.rst, but should have a file for it in docs/explanation/restricted-queues.rst but I don't see it.